pkg/federation/sdlmerge/object_type_extending.go

```diff
@@ -23,15 +26,38 @@ func (e *extendObjectTypeDefinitionVisitor) EnterDocument(operation, _ *ast.Docu
 }
 
 func (e *extendObjectTypeDefinitionVisitor) EnterObjectTypeExtension(ref int) {
-	nodes, exists := e.operation.Index.NodesByNameBytes(e.operation.ObjectTypeExtensionNameBytes(ref))
+	nameBytes := e.operation.ObjectTypeExtensionNameBytes(ref)
+	nodes, exists := e.operation.Index.NodesByNameBytes(nameBytes)
 	if !exists {
 		return
 	}
 
+	hasExtended := false
+	shouldReturn := isQueryOrMutation(nameBytes)
```

Comment:
I'm not sure how else we avoid tripping over queries and mutations (and whether there are other edge cases), but here is a possible solution.